### PR TITLE
N8n 2395 wait node toasts

### PR DESCRIPTION
**Pull request by mutdmour**
- Add toasts to help clarify Wait node behavior
- Support links in toasts
- Refactor modals

### CODE REVIEW

File: packages/editor-ui/src/Interface.ts [hunk 1]
```diff
@@ -577,7 +577,6 @@ export interface IRootState {
 	pushConnectionActive: boolean;
 	saveDataErrorExecution: string;
 	saveDataSuccessExecution: string;
-	saveManualExecutions: boolean;
```
Comment (mutdmour) [inline packages/editor-ui/src/Interface.ts:580]: turns out the value of this is wrong and we are not even using it in workflow settings...

File: packages/editor-ui/src/components/DuplicateWorkflowDialog.vue [hunk 2]
```diff
@@ -3,30 +3,34 @@
...
-					:currentTagIds="currentTagIds"
-					:eventBus="dropdownBus"
-					@blur="onTagsBlur"
-					@esc="onTagsEsc"
-					@update="onTagsUpdate"
-					placeholder="Choose or create a tag"
-					ref="dropdown"
-				/>
-			</el-row>
+			<div :class="$style.content">
```
Comment (mutdmour) [inline packages/editor-ui/src/components/DuplicateWorkflowDialog.vue:12]: just adding some margin here

File: packages/editor-ui/src/components/MainSidebar.vue [hunk 3]
```diff
@@ -3,7 +3,6 @@
 		<about :dialogVisible="aboutDialogVisible" @closeDialog="closeAboutDialog"></about>
 		<executions-list :dialogVisible="executionsListDialogVisible" @closeDialog="closeExecutionsListOpenDialog"></executions-list>
 		<credentials-list :dialogVisible="credentialOpenDialogVisible" @closeDialog="closeCredentialOpenDialog"></credentials-list>
-		<workflow-settings :dialogVisible="workflowSettingsDialogVisible" @closeDialog="closeWorkflowSettingsDialog"></workflow-settings>
```
Comment (mutdmour) [inline packages/editor-ui/src/components/MainSidebar.vue:6]: refactor this modal as to be able to open this directly from toast

File: packages/editor-ui/src/components/Modal.vue [hunk 4]
```diff
@@ -1,63 +1,77 @@
...
 <script lang="ts">
 import Vue from "vue";
 
-const sizeMap: {[size: string]: string} = {
-	xl: '80%',
-	lg: '70%',
-	m: '50%',
-	default: '50%',
-};
-
```
Comment (mutdmour) [inline packages/editor-ui/src/components/Modal.vue:53]: I have given up on trying to standardize these sizes.. much easier to support custom configuration per modal...

File: packages/editor-ui/src/components/Modal.vue [hunk 5]
```diff
@@ -1,63 +1,77 @@
 <template>
-	<div>
-		<el-drawer
```
Comment (mutdmour) [inline packages/editor-ui/src/components/Modal.vue:3]: split drawer into its implementation

File: packages/editor-ui/src/components/Modal.vue [hunk 6]
```diff
@@ -126,60 +136,60 @@ export default Vue.extend({
...
+		flex-direction: column;
+		flex-grow: 1;
+	}
 
-		.modal-content {
-			height: 100%;
-		}
+	.modal-content {
+		overflow: auto;
+		flex-grow: 1;
```
Comment (mutdmour) [inline packages/editor-ui/src/components/Modal.vue:185]: allows for our dream setup for content to collapse while footer remains at bottom... now implemented for workflow settings

File: packages/editor-ui/src/components/mixins/showMessage.ts [hunk 7]
```diff
@@ -33,12 +43,38 @@ export const showMessage = mixins(externalHooks).extend({
...
+					return false;
+				}
+				if (child.parentElement === parent) {
+					return true;
+				}
+
+				return isChildOf(parent, child.parentElement);
+			}
+
+			if (config.onLinkClick) {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/mixins/showMessage.ts:57]: add support for clicks on links within toasts

**Review by mutdmour — COMMENTED** [review]
hints

File: packages/editor-ui/src/components/mixins/pushConnection.ts [hunk 8]
```diff
@@ -218,12 +218,33 @@ export const pushConnection = mixins(
 					// @ts-ignore
 					const workflow = this.getWorkflow();
 					if (runDataExecuted.waitTill !== undefined) {
+						const isNewWorkflow = this.$store.getters.isNewWorkflow;
+						const executionId = this.$store.getters.activeExecutionId;
+						const saveManualExecutions = this.$store.getters.saveManualExecutions;
```
Comment (BHesseldieck) [inline packages/editor-ui/src/components/mixins/pushConnection.ts:223]: I would prefer destruction here, but also fine if you want to keep it as is

File: packages/editor-ui/src/components/mixins/showMessage.ts [hunk 9]
```diff
@@ -33,11 +52,34 @@ export const showMessage = mixins(externalHooks).extend({
 				};
 			}
 
+			if (config.onLinkClick) {
+				const onLinkClick = (e: MouseEvent) => {
+					if (e && e.target && config.onLinkClick && isChildOf(notification.$el, e.target as Element)) {
```
Comment (BHesseldieck) [inline packages/editor-ui/src/components/mixins/showMessage.ts:57]: isn't this checked 2 lines above? Or is TS yelling then?

```suggestion
					if (e && e.target && isChildOf(notification.$el, e.target as Element)) {
```

**Review by BHesseldieck — APPROVED** [review]
Just some small things otherwise 🚀  😊

File: packages/editor-ui/src/components/ModalDrawer.vue [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+			if (event && event.keyCode === 13) {
+				this.handleEnter();
+			}
+		},
+		handleEnter() {
+			if (this.isActive) {
+				this.$emit('enter');
+			}
+		},
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/components/ModalDrawer.vue:67]: cannot merge them. it's also used within the template... depending on where focus is (on window or within the modal), we needed to handle both scenarios

File: packages/editor-ui/src/components/mixins/pushConnection.ts (hunk 8 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/mixins/pushConnection.ts:223]: will keep it as is... easier to search for when looking up store getters.

Comment (mutdmour) [reply packages/editor-ui/src/components/mixins/pushConnection.ts:223]: updated

File: packages/editor-ui/src/components/mixins/showMessage.ts (hunk 9 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/mixins/showMessage.ts:57]: TS yelling 😊

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with n8n@0.139.0

**Pull request merged**